src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,132 @@
+  public usage: string = 'pubSub <subName (optional)> | <"list" | "types" (optional)>';
+  public pluginAlias = ['sub', 'subs', 'sandwich', 'samwich', 'sarnie'];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Food;
+
+  private _API_URL: string = 'https://api.pubsub-api.dev';
+  private _VALID_KEYS: string[] = ['random', 'list', 'types', ''];
+  private _DEFAULT_INPUT: string = 'chicken-tenders';
+  private _SUBS: string[] = [];
+  private _EMBED_LIST = new MessageEmbed();
+
+  private _SUB_UPD_THRESH: number = moment.duration(1, 'days').asMilliseconds();
+  private _LAST_UPD_TIME: number = 0;
 
   constructor(public container: IContainer) {
     super();
+    this._updateData();
+  }
+
+  private async _updateData() {
+    this.container.httpService
+      .get(`${this._API_URL}/allsubs/`)
+      .then((response: IHttpResponse) => {
+        const subs = response.data;
+
+        this._SUBS = subs.map((subData: { name: string }) => subData.name);
+      })
+      .catch((err) => this.container.loggerService.warn(err));
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    const state = this.container.storeService.get('pubSubStore').state;
-    if (state?.onSale) {
-      message.reply('Chicken tender subs are on sale :eyes:');
-    } else {
-      message.reply('Chicken tender subs are **NOT** on sale.');
+    const input = (args || []).join('-').toLowerCase() || this._DEFAULT_INPUT;
+
+    if (input === 'list' || input === 'types') {
+      // Simply return the list of supported subs
+      message.reply(await this._generateEmbedList());
+      return;
+    }
+
+    // checks that the user entered a sub, otherwise it gets random
+    const subType = this._SUBS.find((sub: string) => sub === input) || 'random';
+
+    // recieves the according info and posts, or derps
+    await this.container.httpService
+      .get(`${this._API_URL}/subs/?name=${subType}`)
+      .then((response: IHttpResponse) => {
+        if (Math.floor(response.status / 100) != 2) {
+          message.reply('The API seems to be having some issues at this time.');
+          return;
+        }
+
+        const [subData] = response.data;
+        const embed: MessageEmbed = this._generateEmbedSub(subData, message.guild);
+
+        message.reply(embed);
+      })
+      .catch((err) => this.container.loggerService.warn(err));
+  }
+
+  private async _generateEmbedList(): Promise<MessageEmbed> {
+    const lastListUpdate: number = Date.now() - this._LAST_UPD_TIME;
+
+    if (lastListUpdate < this._SUB_UPD_THRESH) {
+      return this._EMBED_LIST;
     }
+
+    this._LAST_UPD_TIME = Date.now(); // since it only updates once a day, don't need to worry about accuracy
+
+    await this._updateData();
+
+    this._EMBED_LIST.addField('Available Subs', this._SUBS.join('\n'));
+
+    return this._EMBED_LIST;
+  }
```

Comment:
You need to reset the embed, this will constantly append new fields to the same embed